tree/binary.go

```diff
@@ -452,3 +455,110 @@ func (n *node) getHght(h int) int {
 	}
 	return rh //n.right.GetHght(h + 1)
 }
+
+//  commited by prerna 18/01/2022
+// 1. Finding the cousin nodes
+func (t *Tree) IsCousinNode(n1 int, n2 int) bool {
+	if t == nil {
+		return false
+	}
+	return t.root.isCousinNode(t.GetNodeWithValue(n1), t.GetNodeWithValue(n2))
+}
+func (n *node) isCousinNode(n1 *node, n2 *node) bool {
+	if n == nil {
+		return true
+	}
+
+	if n1.height() != n2.height() {
+		return false
+	}
+
+	if (n.left == n1 && n.right == n2) || (n.left == n2 && n.right == n1) {
+		return false
+	}
+	return n.left.isCousinNode(n1, n2) && n.right.isCousinNode(n1, n2)
+}
+
+//2. Printing the top view of a tree
+
+func (t *Tree) PrintTopView() {
+	if t == nil {
+		return
+	}
+	t.root.printTopView()
+}
+
+func (n *node) printTopView() {
+	if n == nil {
+		return
+	}
+	nd := lvlNode{n, 0}
```

Comment:
This is where the type you defined above should be defined. It'll be then a local type.

    type lvlNode struct {
		n1  *node
		lvl int
	}

	nn := &lvlNode{
		n1:  n,
		lvl: 0,
	}

	q := queue.New()
	q.Enqueue(nn)
	m := make(map[int]bool)
	for q.Len() != 0 {
		n := q.Dequeue().(*lvlNode)
		if _, ok := m[n.lvl]; !ok {
			fmt.Printf("%d\t", n.n1.value)
			m[n.lvl] = true
		}

		if n.n1.left != nil {
			nn = &lvlNode{
				n1:  n.n1.left,
				lvl: n.lvl + 1,
			}
			q.Enqueue(nn)
		}

		if n.n1.right != nil {
			nn = &lvlNode{
				n1:  n.n1.right,
				lvl: n.lvl - 1,
			}
			q.Enqueue(nn)
		}
	}

Something like this.